pkg/component/kubernetes/apiserver/deployment.go

```diff
@@ -614,6 +635,33 @@ func (k *kubeAPIServer) handleVPNSettingsHA(
 		})
 	}
 
+	// Only inject Envoy Proxy if Seed and Shoot networks overlap. We need it for the network mapping.
+	overlap := netutils.OverLapAny(k.values.VPN.SeedPodNetwork, slices.Concat(k.values.VPN.PodNetworkCIDRs, k.values.ServiceNetworkCIDRs, k.values.VPN.NodeNetworkCIDRs)...)
+	if overlap {
+		deployment.Spec.Template.Spec.Containers[0].Args = append(deployment.Spec.Template.Spec.Containers[0].Args, fmt.Sprintf("--egress-selector-config-file=%s/%s", volumeMountPathEgressSelector, configMapEgressSelectorDataKey))
+		deployment.Spec.Template.Spec.HostAliases = append(deployment.Spec.Template.Spec.HostAliases, corev1.HostAlias{
+			IP: "127.0.0.1",
+			Hostnames: []string{
+				EnvoyHostHAVPN,
+			},
+		})
```

Comment:
I think we should be good without this when we use `localhost` (see above). Did I miss something?